test/unit/core.ts

```diff
@@ -837,6 +838,44 @@ describe("snabbdom", function () {
           assert.equal(elm.children[1].innerHTML, "symbol");
           assert.equal(elm.children[2].innerHTML, "symbol");
         });
+        it("simultaneous addition in beginning and removal in end", function () {
+          let insertCount = 0;
+          const domApi: DOMAPI = {
+            ...htmlDomApi,
+            insertBefore: (a, b, c) => {
+              insertCount++;
+              htmlDomApi.insertBefore(a, b, c);
+            }
+          };
+          const patch = init([], domApi);
+          const vnode1 = h("span", [2, 3, 4, 5].map(spanNum));
+          const vnode2 = h("span", [1, 2, 3, 4].map(spanNum));
+          elm = patch(vnode0, vnode1).elm;
+          assert.deepEqual(map(inner, elm.children), ["2", "3", "4", "5"]);
+          elm = patch(vnode1, vnode2).elm;
+          assert.deepEqual(map(inner, elm.children), ["1", "2", "3", "4"]);
+          assert.strictEqual(insertCount, 1);
+        });
+        it("simultaneous removal in beginning and addition in end", function () {
+          let insertCount = 0;
+          const domApi: DOMAPI = {
+            ...htmlDomApi,
+            insertBefore: (a, b, c) => {
+              insertCount++;
+              htmlDomApi.insertBefore(a, b, c);
+            }
+          };
```

Comment:
This change has no effect on the resulting DOM, only on the process by which we get there. Hence the test wraps the `htmlDomApi` to be able to observe what goes on in there.